api/client/image/import.go

```diff
@@ -0,0 +1,94 @@
+package image
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/api/client"
+	"github.com/docker/docker/cli"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/urlutil"
+	"github.com/docker/engine-api/types"
+	"github.com/spf13/cobra"
+)
+
+type importOptions struct {
+	source    string
+	reference string
+	changes   []string
+	message   string
+
+	// deprecated
+	tag string
+}
+
+// NewImportCommand creates a new `docker import` command
+func NewImportCommand(dockerCli *client.DockerCli) *cobra.Command {
+	var opts importOptions
+
+	cmd := &cobra.Command{
+		Use:   "import [OPTIONS] file|URL|- [REPOSITORY[:TAG]]",
+		Short: "Import the contents from a tarball to create a filesystem image",
+		Args:  cli.RequiresMinArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.source = args[0]
+			opts.reference = args[1]
```

Comment:
min args is set to 1, but it looks like this assumes 2.

I think this needs `if len(args) > 1`